File: mario/more/mario.c
// Recreates Mario pyramids using hashes

#include <cs50.h>
#include <stdio.h>

int get_input(void);
void create_pyramid(int height);

int main(void)
{
    int height = get_input();
    create_pyramid(height);
}

// Prompts the user for a height between 1 and 8, inclusive
int get_input(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);
    return height;
}

// Prints n number of hashes
void print_hash(int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("#");
    }
}

// Creates a pyramid of hashes of given height
void create_pyramid(int height)
{
    for (int line = 1; line <= height; line++)
    {
        // Prints (height - line) number of space characters
        for (int i = line; i < height; i++)
        {
            printf(" ");
        }
        print_hash(line);
        printf("  ");
        print_hash(line);
        printf("\n");
    }
}

File: mario/more/mario.py
from cs50 import get_int


def get_positive_int(prompt):
    # Function to get positive integer not greater than 8
    while True:
        height = get_int(prompt)
        if height > 0 and height <= 8:
            return height


def main():
    # Prompt user for pyramid height
    height = get_positive_int("Height: ")

    # Generate pyramid
    for line in range(1, height + 1):

        # Print (height - line) number of leading white space characters
        print(" " * (height - line), end="")

        # Print line number of hashes (#)
        print("#" * line, end="")

        # Print 2 white space characters
        print("  ", end="")

        # Print line number of hashes (#) again
        print("#" * line)


if __name__ == "__main__":
    main()